### PR TITLE
refactor(GraphQL): Pointer constraint input type as ID

### PR DESCRIPTION
**Pull request by douglasmuraoka**
Redefines the Pointer constraint input type from a custom scalar to
a simple ID.

### CODE REVIEW

File: src/GraphQL/loaders/parseClassTypes.js [hunk 1]
```diff
@@ -341,10 +235,14 @@ const load = (
     name: classGraphQLConstraintTypeName,
     description: `The ${classGraphQLConstraintTypeName} input type is used in operations that involve filtering objects by a pointer field to ${graphQLClassName} class.`,
     fields: {
-      _eq: defaultGraphQLTypes._eq(classGraphQLScalarType),
-      _ne: defaultGraphQLTypes._ne(classGraphQLScalarType),
-      _in: defaultGraphQLTypes._in(classGraphQLScalarType),
-      _nin: defaultGraphQLTypes._nin(classGraphQLScalarType),
+      _eq: defaultGraphQLTypes._eq(GraphQLID),
+      _ne: defaultGraphQLTypes._ne(GraphQLID),
+      _in: defaultGraphQLTypes._in(
```
Comment (davimacedo) [inline src/GraphQL/loaders/parseClassTypes.js]: I think it should be `in: defaultGraphQLTypes._in(defaultGraphQLTypes.OBJECT_ID)`

File: src/GraphQL/loaders/parseClassTypes.js [hunk 2]
```diff
@@ -341,10 +235,14 @@ const load = (
...
+        new GraphQLList(defaultGraphQLTypes.OBJECT_ID)
+      ),
+      _nin: defaultGraphQLTypes._nin(
```
Comment (davimacedo) [inline src/GraphQL/loaders/parseClassTypes.js]: and here: `_nin: defaultGraphQLTypes._nin(defaultGraphQLTypes.OBJECT_ID)`

File: src/GraphQL/loaders/parseClassTypes.js [hunk 3]
```diff
@@ -341,10 +235,14 @@ const load = (
...
+        new GraphQLList(defaultGraphQLTypes.OBJECT_ID)
+      ),
       _exists: defaultGraphQLTypes._exists,
       _select: defaultGraphQLTypes._select,
```
Comment (davimacedo) [inline src/GraphQL/loaders/parseClassTypes.js]: I think we should also improve these following 4 operators. But we can address it in a separate PR. I will take a note in the project.

File: src/GraphQL/loaders/parseClassTypes.js [hunk 4]
```diff
@@ -284,30 +202,6 @@ const load = (
     classGraphQLUpdateType
   );
 
-  const classGraphQLPointerTypeName = `${graphQLClassName}PointerInput`;
```
Comment (davimacedo) [inline src/GraphQL/loaders/parseClassTypes.js]: Why are you removing this one?

Comment (douglasmuraoka) [reply src/GraphQL/loaders/parseClassTypes.js]: That was a mistake, thanks for noticing it. It was supposed to remove only the Pointer scalar type.

**Pull request merged**